### ISSUE TITLE
Upgrade to Python 3

### ISSUE THREAD

**Issue by domenic** [tabatkins/highlighter#16]
This is a dependency of https://github.com/whatwg/html-build/issues/220.

**Comment by tabatkins** [comment]
All right, initial commit up. @sideshowbarker, wanna check that it's good for you? As far as I can tell it works for me!

**Comment by sideshowbarker** [comment]
OK I did some follow-up in https://github.com/tabatkins/highlighter/pull/17, and with the changes from that PR applied, highlighting of the HTML spec works as expected under Python 3.

(Sorry, it took be a little longer than I had expected to get it done; it’s because IDL highlighting was failing completely — but due to the nature of that code, no exceptions get thrown when it’s not actually doing anything. Instead, the important part of it just never gets run. So it took me a while to figure out what needed to get changed to make it work again.)

**Comment by sideshowbarker** [comment]
FYI about the cause of the IDL highlighting problem I mentioned in my previous comment here, I filed https://github.com/plinss/widlparser/issues/59. This seems like maybe a good case for using an abstract class — which from https://docs.python.org/3/library/abc.html I notice Python 3 now has.

But maybe (preferably) there’s an existing native Pythonic way I’m so far unaware of that could be used/required by writers of consumer code to avoid the kind of problem I ran into.
